x/domain/keeper/executor/stores.go

```diff
@@ -0,0 +1,185 @@
+package executor
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/iov-one/iovns/pkg/index"
+	"github.com/iov-one/iovns/x/domain/types"
+)
+
+// IndexStorePrefix is the prefix used to defines the prefixed store containing indexing data
+var IndexStorePrefix = []byte{0x02}
+
+// DomainStorePrefix is the prefix used to define the prefixed store containing prefixedStore data
+var DomainStorePrefix = []byte{0x00}
+
+// AccountPrefixStore is the prefix used to define the prefixed store containing account data
+var AccountStorePrefix = []byte{0x01}
+var ownerToDomainPrefix = []byte{0x05}
+var ownerToAccountPrefix = []byte{0x04}
+var blockchainTargetsPrefix = []byte{0x06}
+
+type indexStore struct {
+	store sdk.KVStore
+}
+
+func newIndexStore(ctx sdk.Context, key sdk.StoreKey) indexStore {
+	store := ctx.KVStore(key)
+	store = prefix.NewStore(store, IndexStorePrefix)
+	return indexStore{
+		store: store,
+	}
+}
+
+func (i indexStore) ownerToDomain(addr sdk.AccAddress) (index.Store, error) {
+	if addr.Empty() {
+		panic("indexing of empty addresses is not allowed")
+	}
+	return index.NewAddressIndex(i.store, ownerToDomainPrefix, addr)
+}
+
+func (i indexStore) ownerToAccount(addr sdk.AccAddress) (index.Store, error) {
+	if addr.Empty() {
+		panic("indexing of empty addresses is not allowed")
+	}
+	return index.NewAddressIndex(i.store, ownerToAccountPrefix, addr)
+}
+
+func (i indexStore) targetToAccount(target types.BlockchainAddress) (index.Store, error) {
+	return index.NewIndexedStore(i.store, blockchainTargetsPrefix, target)
+}
+
+type accountStore struct {
+	cdc      *codec.Codec
+	store    sdk.KVStore
+	idxStore indexStore
+}
+
+func newAccountStore(ctx sdk.Context, key sdk.StoreKey, cdc *codec.Codec) accountStore {
+	return accountStore{
+		store:    prefix.NewStore(ctx.KVStore(key), AccountStorePrefix),
+		idxStore: newIndexStore(ctx, key),
+		cdc:      cdc,
+	}
+}
+
+// create creates a new account in the account store
+func (a accountStore) create(account types.Account) {
```

Comment:
Let's create:
```go
type CRUDStore interface {
  Create()
  Read()
  Update()
  Delete()
}
```
If possible. I know we need to do some refactoring on account/domain types and wrap them in interfaces. This improvement could be considered in the next sprints. What do you think @fdymylja would it make a better API?

Reply:
Agreed, today I'm trying to achieve this type of CRUDStore. My idea right now would be to wrap the types (Account, Domain) for now and implement an interface that allows them to be "crudded", I'm wrapping them for now in order to avoid to have too many merge conflicts. Then we could discuss of removing the wrapping and implement the "CRUDDABLE" interface onto the types directly.